workspaces/ui-v2/src/lib/__tests/diff-helpers/fixture.ts

```diff
@@ -1,282 +1,164 @@
+import { IShapeTrail } from '@useoptic/cli-shared/build/diffs/shape-trail';
+import colors from 'colors';
+import {
+  ICopy,
+  ICopyStyle,
+} from '<src>/optic-components/diffs/render/ICopyRender';
+import { IChangeType, IInterpretation } from '<src>/lib/Interfaces';
+import { IValueAffordanceSerializationWithCounter } from '@useoptic/cli-shared/build/diffs/initial-types';
+
+export const testCase = (basePath: string) => async (
+  name: string
+): Promise<{
+  universe: ITestUniverse;
+  diffSet: DiffSet;
+  diffs: ParsedDiff[];
+}> => {
+  const universe = await buildUniverse(
+    require(path.join(__dirname + '/universes', basePath, name + '.json'))
+  );
+
+  const started = await universe.opticContext.capturesService.startDiff(
+    '123',
+    'example-session'
+  );
+  await started.onComplete;
+
+  const result = await universe.opticContext.diffRepository.findById('123');
+  const diffs = (await result.listDiffs()).diffs;
+
+  const parsedDiffs = diffs.map(
+    ([diff, interactions, fingerprint]: any) =>
+      new ParsedDiff(diff, interactions, fingerprint)
+  );
+
+  return {
+    diffs: parsedDiffs,
+    diffSet: new DiffSet(parsedDiffs, universe.currentSpecContext),
+    universe,
+  };
+};
+export const testCaseLoaded = async (
+  example: any
+): Promise<{
+  universe: ITestUniverse;
+  diffSet: DiffSet;
+  diffs: ParsedDiff[];
+}> => {
+  const universe = await buildUniverse(example);
+
+  const started = await universe.opticContext.capturesService.startDiff(
+    '123',
+    'example-session'
+  );
+  await started.onComplete;
+
+  const result = await universe.opticContext.diffRepository.findById('123');
+  const diffs = (await result.listDiffs()).diffs;
+
+  const parsedDiffs = diffs.map(
+    ([diff, interactions, fingerprint]: any) =>
+      new ParsedDiff(diff, interactions, fingerprint)
+  );
+
+  return {
+    diffs: parsedDiffs,
+    diffSet: new DiffSet(parsedDiffs, universe.currentSpecContext),
+    universe,
+  };
+};
+
+export type IShapeDiffTestSnapshot = {
+  preview: IInterpretation;
+  commands: any[];
+  trailValues: IValueAffordanceSerializationWithCounter;
+};
+
+export async function shapeDiffPreview(
+  input: {
+    shapeDiffGroupingHash: string;
+    shapeTrail: IShapeTrail;
+    diffs: ParsedDiff[];
+  },
+  universe: ITestUniverse
+): Promise<IShapeDiffTestSnapshot> {
+  const { pathId, method } = input.diffs[0].location(
+    universe.currentSpecContext
+  );
+
+  const diffService = await universe.opticContext.diffRepository.findById(
+    '123'
+  );
+  const shapeAffordances = await diffService.learnShapeDiffAffordances();
+
+  const trailValues = shapeAffordances[input.diffs[0]!.diffHash];
+
+  const preview = await interpretShapeDiffs(
+    input.diffs[0].asShapeDiff(universe.currentSpecContext)!,
+    trailValues,
+    universe.spectacleQuery,
+    universe.currentSpecContext
+  );
+
+  return {
+    preview,
+    trailValues,
+    commands: [], //preview.toCommands(preview.updateSpecChoices!),
```

Comment:
@devdoshi  -- if you wanted to make the tests generate commands / prompt that WASM issue, here's where that happens